File: frontAnimalCrossing/src/app/models/bugCatched.ts
import { IslandLeader } from "./islandLeader";

export class BugCatched{


  constructor(
  private _id: number | null,
  private _name: string,
  private _islandLeader: IslandLeader,
  private _bugCatched: string[]
  ){}
  public get bugCatched(): string[] {
    return this._bugCatched;
  }
  public set bugCatched(value: string[]) {
    this._bugCatched = value;
  }
  public get islandLeader(): IslandLeader {
    return this._islandLeader;
  }
  public set islandLeader(value: IslandLeader) {
    this._islandLeader = value;
  }
  public get name(): string {
    return this._name;
  }
  public set name(value: string) {
    this._name = value;
  }
  public get id(): number | null {
    return this._id;
  }
  public set id(value: number | null) {
    this._id = value;
  }

  public toJSON(): any {
    return{
      id:this._id,
      name:this._name,
      islandLeader:this._islandLeader,
      bugCatched:this._bugCatched

    }
}
}

File: backAnimalCrossing/src/main/java/com/ironhack/backAnimalCrossing/model/BugCatched.java
package com.ironhack.backAnimalCrossing.model;

import javax.persistence.*;

@Entity
public class BugCatched {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToOne
    @JoinColumn(name = "islandLeader_Bid")
    private IslandLeader islandLeader;

    private String[] bugs;

    public BugCatched() {
    }

    public BugCatched(String name, IslandLeader islandLeader, String[] bugs) {
        this.name = name;
        this.islandLeader = islandLeader;
        this.bugs = bugs;
    }

    public BugCatched(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public IslandLeader getIslandLeader() {
        return islandLeader;
    }

    public void setIslandLeader(IslandLeader islandLeader) {
        this.islandLeader = islandLeader;
    }

    public String[] getBugs() {
        return bugs;
    }

    public void setBugs(String[] bugs) {
        this.bugs = bugs;
    }
}
